tests/test_plugin.py

```diff
@@ -931,3 +931,27 @@ def test_duplicate_plugin_names():
     # NOTE: This assumes the test succeeds. If it fails, then
     # subsequent tests can fail because of it.
     pyblish.plugin.ALLOW_DUPLICATES = False
+
+
+@with_setup(lib.setup_empty, lib.teardown)
+def test_validate_publish_data_member_type():
+    """Validate publish data member type works."""
+
+    pyblish.plugin.STRICT_DATATYPES = True
```

Comment:
..Although, that would limit your ability to do this.

Let's leave it. Looks good!